src/ir/objc.rs

```diff
@@ -135,14 +139,23 @@ impl ObjCInterface {
             interface.is_protocol = true;
         }
 
+        // This is the ItemId for the real interface to which a category extends.  We must make it
+        // an optional, set it when we visit the ObjCCategoryDecl and then use it after we've
+        // visited the entire tree. We must do it in this order to ensure that this interface has
+        // all the template tags assigned to it.
+        let mut real_interface_id_for_category: Option<ItemId> = None;
+
         cursor.visit(|c| {
             match c.kind() {
                 CXCursor_ObjCClassRef => {
                     if cursor.kind() == CXCursor_ObjCCategoryDecl {
                         // We are actually a category extension, and we found the reference
-                        // to the original interface, so name this interface approriately
+                        // to the original interface, so name this interface approriately.
+
                         interface.name = c.spelling();
                         interface.category = Some(cursor.spelling());
+                        real_interface_id_for_category = Some(Item::from_ty_or_ref(c.cur_type(), c, None, ctx).into());
```

Comment:
Does objective-c have the guarantee that the category  needs to be defined before?

Otherwise you have no guarantee to find it in the map.

Reply:
Yes. It does guarantee it. I'm glad you made me check. I just kind of assumed it would be the case.

Clang throws an error with the following:
```objc
@interface Foo (BarCategory)
@end 

@interface Foo
@end 
```

but is fine with:
```objc
@interface Foo                                                                                                                                                                                                                                                                   
@end                                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                 
@interface Foo (BarCategory)                                                                                                                                                                                                                                                     
@end 
```